### ISSUE TITLE
different results with same data anda parameters (R and Python)

### ISSUE THREAD

**Issue by IanniMuliterno** [dmlc/xgboost#5767]
Hello there,

 I spotted a problem  when training a xgboost (version ‘0.82.1’) in R 2.6.2 (with RStudio 1.2.1335) and comparing with xgboost (version '1.1.0') in python 3.7.6 (with Spyder 4.0.1).
Even tough I replicate the setup, I get very different results, also in python as I allow the model to have more trees, the feature importance changes while the same doesn't happen in R (R is much more stable). What would the reason be?  Shouldn't the algorithm be the same?
I wondered if version is the problem, but if that is the case, I believe the differences wouldn't be so significant.

**Comment by trivialfis** [comment]
Could you please provide the parameters you used?

**Comment by IanniMuliterno** [comment]
Of course, here it is.

R setup:

                  max_depth = 3,
                  eta = 0.12,
                  gamma = 0,
                  colsample_bytree = 0.6,
                  subsample = 0.7,
                  scale_pos_weight = 6,
                  eval_metric = 'auc',
                  objective = 'binary:logistic'
                  ntrees = 45

Python setup:

                   max_depth = 3,
                   learning_rate = 0.12,
                   gamma = 0,
                   colsample_bytree = 0.6,
                   subsample = 0.7,
                   scale_pos_weight = 6,
                   eval_metric = 'auc',
                   objective = 'binary:logistic',
                   n_estimators = 45

**Comment by IanniMuliterno** [comment]
A friend is loading at Python the same version of the package that was used in R, I will report any interesting changes in the outcome.

**Comment by trivialfis** [comment]
I bisected the code, the change is mostly due to b833b642eca0 .  That commit changes the default seed of the algorithm.  R package uses random engine from R instead of the one in c++ so it might seem more stable.

**Comment by trivialfis** [comment]
I don't think it's a bug here as the cause roots from change of random engine instead of boosting algorithm.  Feel free to reopen if you have further information.
